composite/bitcoin-json-rpc/src/adapter.ts

```diff
@@ -0,0 +1,44 @@
+import JSONRPC from '@chainlink/json-rpc-adapter'
+import { Execute, AdapterRequest } from '@chainlink/types'
+import { Validator } from '@chainlink/external-adapter'
+import { Requester } from '@chainlink/external-adapter'
+
+const inputParams = {
+  url: false,
+  method: false,
+  params: false,
+  blockchain: false,
+  coin: false,
+  endpoint: false,
+}
+
+const convertEndpoint: { [key: string]: string } = {
+  height: 'headers',
+}
+
+// Export function to integrate with Chainlink node
+export const execute: Execute = async (request: AdapterRequest) => {
+  const validator = new Validator(request, inputParams)
+  const blockchain = validator.validated.data.blockchain || validator.validated.data.coin
+
+  let endpoint = validator.validated.data.endpoint
+
+  if (validator.error) throw validator.error
+  if (endpoint != undefined || blockchain != undefined) {
+    if (endpoint in convertEndpoint) endpoint = convertEndpoint[endpoint]
+    if (!endpoint) endpoint = 'difficulty'
+    request.data.method = 'getblockchaininfo'
+  }
+
+  const response = await JSONRPC.execute(request)
+
+  if (endpoint) {
```

Comment:
I think you are repeating the same logic as before here? `endpoint` will be defined already right?